C/decoder/decode.c

```diff
@@ -0,0 +1,215 @@
+#include "decode.h"
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+static char chars[83] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz#$%*+,-.:;=?@[]^_{|}~";
```

Comment:
@lautat , I looked at the function and variable definitions from `encode.c`, but to have an identical definitions of these functions/symbols, we have to link `decode.o` with `encode.o` during linking phase, so that the same symbols will be used. But if you look at the `makefile`, encoder and decoders are separate targets, and also `encode.h` doesn't export these functions in its header file, i.e `encode.h`, so this will involve changing `encode.c`, `encode.h` as well. If you are okay with that, I can make it.

Reply:
Yes, you can modify `encode.c` and `encode.h` if you need to. I think you can create a common header file to include declarations for these functions and optionally definitions. Keeping encoder and decoder separate sounds like a good idea to me.

Reply:
@lautat 
Okay, thanks. I will create `common.c` and `common.h` that defines and declares these common functions.

Another idea would be to create only `common.h` and make these functions as `static inline`, so that these functions will just be inline definitions and will be substituted back in their original places during compilation. What are your views on this??

Reply:
> Another idea would be to create only common.h and make these functions as static inline, so that these functions will just be inline definitions and will be substituted back in their original places during compilation. What are your views on this??

This is also a good option. I recall that this approach would allow compilers to apply more optimizations.

Reply:
Yes. So can I make them `static inline` ??

Reply:
Yes, you can :+1:

Reply:
I have made the necessary changes to the repo. Also I have kept `chars` array in `decode.c` as it is, since declaring it in the header will not make sense and adds complexity.